### ISSUE TITLE
Compilation failure.

### ISSUE THREAD

**Issue by Achaean** [juzzlin/DustRacing2D#32]
On Debian Jessie (oldstable) x64 KDE.
Here are the last lines from make:

```
[ 57%] Building CXX object src/game/CMakeFiles/dustrac-game.dir/statemachine.cpp.o
/home/giorgos/TEMPG/DustRacing2D-2.0.1/src/game/statemachine.cpp: In member function 'void StateMachine::stateDoIntro()':
/home/giorgos/TEMPG/DustRacing2D-2.0.1/src/game/statemachine.cpp:135:6: error: no matching function for call to 'QTimer::singleShot(int, StateMachine::stateDoIntro()::<lambda()>)'
     });
      ^
/home/giorgos/TEMPG/DustRacing2D-2.0.1/src/game/statemachine.cpp:135:6: note: candidates are:
In file included from /usr/include/x86_64-linux-gnu/qt5/QtCore/QTimer:1:0,
                 from /home/giorgos/TEMPG/DustRacing2D-2.0.1/src/game/eventhandler.hpp:23,
                 from /home/giorgos/TEMPG/DustRacing2D-2.0.1/src/game/renderer.hpp:22,
                 from /home/giorgos/TEMPG/DustRacing2D-2.0.1/src/game/statemachine.hpp:19,
                 from /home/giorgos/TEMPG/DustRacing2D-2.0.1/src/game/statemachine.cpp:16:
/usr/include/x86_64-linux-gnu/qt5/QtCore/qtimer.h:81:17: note: static void QTimer::singleShot(int, const QObject*, const char*)
     static void singleShot(int msec, const QObject *receiver, const char *member);
                 ^
/usr/include/x86_64-linux-gnu/qt5/QtCore/qtimer.h:81:17: note:   candidate expects 3 arguments, 2 provided
/usr/include/x86_64-linux-gnu/qt5/QtCore/qtimer.h:82:17: note: static void QTimer::singleShot(int, Qt::TimerType, const QObject*, const char*)
     static void singleShot(int msec, Qt::TimerType timerType, const QObject *receiver, const char *member);
                 ^
/usr/include/x86_64-linux-gnu/qt5/QtCore/qtimer.h:82:17: note:   candidate expects 4 arguments, 2 provided
src/game/CMakeFiles/dustrac-game.dir/build.make:905: recipe for target 'src/game/CMakeFiles/dustrac-game.dir/statemachine.cpp.o' failed
make[2]: *** [src/game/CMakeFiles/dustrac-game.dir/statemachine.cpp.o] Error 1
CMakeFiles/Makefile2:538: recipe for target 'src/game/CMakeFiles/dustrac-game.dir/all' failed
make[1]: *** [src/game/CMakeFiles/dustrac-game.dir/all] Error 2
Makefile:160: recipe for target 'all' failed
make: *** [all] Error 2
>
```

**Comment by juzzlin** [comment]
What is your compiler and version? What is your Qt version? The current master branch builds just fine on Ubuntu 16.04 and Ubuntu 18.04 against the default Qt (5.6.x and 5.9.x). 5.10 is also tested.

**Comment by juzzlin** [comment]
After a quick investigation the minimum required Qt version seems to be too low at the moment (5.2.1). At least Qt 5.4 is required because of this version of QTimer::singleShot().

**Comment by Achaean** [comment]
Oh, that\s explains it! :-)

I have v.5.3.2. I'll try again as soon as I'll get my hands at a newer QT version.
Bye! :-)
A.

**Comment by juzzlin** [comment]
The version check is now fixed in fb1f68512486060373507d2d9e3faed363aa9099

Thanks for reporting! :)
